apps/design/backend/src/store.ts

```diff
@@ -352,4 +407,73 @@ export class Store {
       electionId
     );
   }
+
+  //
+  // Background task processing
+  //
+
+  getQueuedBackgroundTasks(): BackgroundTask[] {
+    const sql = `${getBackgroundTasksBaseQuery}
+      where started_at is null
+      order by created_at asc
+    `;
+    const rows = this.client.all(sql) as BackgroundTaskRow[];
+    return rows.map(backgroundTaskRowToBackgroundTask);
+  }
+
+  getBackgroundTask(taskId: Id): Optional<BackgroundTask> {
+    const sql = `${getBackgroundTasksBaseQuery}
+      where id = ?
+    `;
+    const row = this.client.one(sql, taskId) as Optional<BackgroundTaskRow>;
+    return row ? backgroundTaskRowToBackgroundTask(row) : undefined;
+  }
+
+  createBackgroundTask(taskName: string, payload: unknown): Id {
+    const taskId = uuid();
+    this.client.run(
+      `
+      insert into background_tasks (id, task_name, payload)
+      values (?, ?, ?)
+      `,
+      taskId,
+      taskName,
+      JSON.stringify(payload)
+    );
+    return taskId;
+  }
+
+  startBackgroundTask(taskId: Id): void {
+    this.client.run(
+      `
+      update background_tasks
+      set started_at = current_timestamp
+      where id = ?
+      `,
+      taskId
+    );
+  }
+
+  completeBackgroundTask(taskId: Id, error?: string): void {
+    if (error) {
+      this.client.run(
+        `
+        update background_tasks
+        set completed_at = current_timestamp, error = ?
+        where id = ?
+        `,
+        error,
+        taskId
+      );
```

Comment:
What happens if you run this case when `error` is undefined? Wondering if it would just work

Reply:
Ah good catch! Can indeed consolidate (with a conversion of `undefined` to `null`): https://github.com/votingworks/vxsuite/pull/4162/commits/d26a754961434311ec1e46058f53499b6a3b24be